src/Membrain/Memory.hs

```diff
@@ -104,6 +110,12 @@ instance Monoid (Memory (mem :: Nat)) where
     mconcat = foldl' (<>) mempty
     {-# INLINE mconcat #-}
 
+#if ( __GLASGOW_HASKELL__ >= 804 )
+type IsMemory mem = (KnownNat mem, KnownUnitSymbol mem, Terminating mem)
+#else
+type IsMemory mem = (KnownNat mem, KnownUnitSymbol mem)
```

Comment:
I can be a bit picky, but I would call it `KnownMem` for better description of what this alias is.